internal/provider/resource_aci_pim_route_map_entry.go

```diff
@@ -0,0 +1,595 @@
+			}
+		}
+	} else {
+		data.TagAnnotation = types.SetNull(data.TagAnnotation.ElementType(ctx))
+	}
+
+	return childPayloads, "", ""
+}
+
+func getPimRouteMapEntryCreateJsonPayload(ctx context.Context, data *PimRouteMapEntryResourceModel, tagAnnotationPlan, tagAnnotationState []TagAnnotationPimRouteMapEntryResourceModel) (*container.Container, string, string) {
+	payloadMap := map[string]interface{}{}
+	payloadMap["attributes"] = map[string]string{}
+	childPayloads := []map[string]interface{}{}
+
+	TagAnnotationchildPayloads, errMessage, errMessageDetail := getPimRouteMapEntryTagAnnotationChildPayloads(ctx, data, tagAnnotationPlan, tagAnnotationState)
+	if TagAnnotationchildPayloads == nil {
+		return nil, errMessage, errMessageDetail
+	}
+	childPayloads = append(childPayloads, TagAnnotationchildPayloads...)
+
+	payloadMap["children"] = childPayloads
+	if !data.Action.IsNull() && !data.Action.IsUnknown() {
+		payloadMap["attributes"].(map[string]string)["action"] = data.Action.ValueString()
+	}
+	if !data.Annotation.IsNull() && !data.Annotation.IsUnknown() {
+		payloadMap["attributes"].(map[string]string)["annotation"] = data.Annotation.ValueString()
+	}
+	if !data.Descr.IsNull() && !data.Descr.IsUnknown() {
+		payloadMap["attributes"].(map[string]string)["descr"] = data.Descr.ValueString()
+	}
+	if !data.Grp.IsNull() && !data.Grp.IsUnknown() {
+		payloadMap["attributes"].(map[string]string)["grp"] = data.Grp.ValueString()
+	}
+	if !data.Name.IsNull() && !data.Name.IsUnknown() {
+		payloadMap["attributes"].(map[string]string)["name"] = data.Name.ValueString()
+	}
+	if !data.NameAlias.IsNull() && !data.NameAlias.IsUnknown() {
+		payloadMap["attributes"].(map[string]string)["nameAlias"] = data.NameAlias.ValueString()
+	}
+	if !data.Order.IsNull() && !data.Order.IsUnknown() {
+		payloadMap["attributes"].(map[string]string)["order"] = data.Order.ValueString()
+	}
+	if !data.Rp.IsNull() && !data.Rp.IsUnknown() {
+		payloadMap["attributes"].(map[string]string)["rp"] = data.Rp.ValueString()
+	}
+	if !data.Src.IsNull() && !data.Src.IsUnknown() {
+		payloadMap["attributes"].(map[string]string)["src"] = data.Src.ValueString()
+	}
+
+	payload, err := json.Marshal(map[string]interface{}{"pimRouteMapEntry": payloadMap})
+	if err != nil {
+		errMessage := "marshalling of json payload failed"
+		errMessageDetail := fmt.Sprintf("err: %s. Please report this issue to the provider developers.", err)
+		return nil, errMessage, errMessageDetail
+	}
+
+	jsonPayload, err := container.ParseJSON(payload)
+
+	if err != nil {
+		errMessage := "construction of json payload failed"
+		errMessageDetail := fmt.Sprintf("err: %s. Please report this issue to the provider developers.", err)
+		return nil, errMessage, errMessageDetail
+	}
+	return jsonPayload, "", ""
+}
+
+func getPimRouteMapEntryDeleteJsonPayload(ctx context.Context, data *PimRouteMapEntryResourceModel) (*container.Container, string, string) {
```

Comment:
Would it be good to have just one Delete function that can be shared across all the resources?

Reply:
I kept it separated for trouble shooting, and if deviations are required for a specific class in future. 

Also not all classes are allowed to be deleted, thus they would run a different set of code simply removing from state. We could add logic for this in the template to call the correct generic function.

Since it is auto-generated code I thought that it would not hurt defining also for every module. We could create a manual utils functions for these functions ( basically replacing the template part ).

Reply:
to be done at later stage